flycheck.el

```diff
@@ -2159,10 +2162,43 @@ into the verification results."
         (princ (make-string (- message-column (current-column)) ?\ ))
         (let ((message (flycheck-verification-result-message result))
               (face (flycheck-verification-result-face result)))
-          (insert (propertize message 'face face)))
+          ;; If face is nil, using propertize erases the face already contained
+          ;; by the message.  We don't want that, since this would remove the
+          ;; button face from the checker chain result.
+          (if face
+              (insert (propertize message 'face face))
+            (insert message)))
         (princ "\n"))))
   (princ "\n"))
 
+(defun flycheck-checker-chain (checker)
+  "Return the chain of checkers following CHECKER.
+
+The return value is a list of checkers, not including CHECKER."
+  (let ((chain))
+    (while (setq checker (flycheck-get-next-checker-for-buffer checker))
+      (push checker chain))
+    (nreverse chain)))
+
+(defun flycheck--verify-checker-chain (checker)
+  "Return a verification result for the checker chain starting from CHECKER."
+  (-when-let (chain (flycheck-checker-chain checker))
+    (list
+     (flycheck-verification-result-new
+      :label "next checkers"
+      ;; We use `insert-text-button' to preserve the button properties in
+      ;; the string
+      :message (string-join
+                (split-string
+                 (with-temp-buffer
+                   (dolist (checker chain)
+                     (insert-text-button (symbol-name checker)
+                                         'type 'help-flycheck-checker-doc
+                                         'help-args (list checker))
+                     (insert " "))
+                   (buffer-string)))
+                ", ")))))
```

Comment:
Cool trick: `make-text-button` can take a string and propertize it; hence this should work:

```elisp
 (string-join
  (seq-map (lambda (checker)
             (make-text-button (symbol-name checker) nil
                               'type 'help-flycheck-checker-doc
                               'help-args (list checker)))
           chain)
  ", ")
```